vowpalwabbit/core/include/vw/core/input_parser.h

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) by respective owners including Yahoo!, Microsoft, and
+// individual contributors. All rights reserved. Released under a BSD (revised)
+// license as described in the file LICENSE.
+#pragma once
+
+#include "vw/common/string_view.h"
+#include "vw/core/vw_fwd.h"
+
+#include <string>
+
+namespace VW
+{
+namespace details
+{
+// Experimental.
+struct input_parser
+{
+  input_parser(std::string name) : _name(std::move(name)) {}
+  virtual ~input_parser() = default;
+
+  virtual bool next(VW::workspace& workspace_instance, io_buf& buffer, VW::v_array<VW::example*>& output_examples) = 0;
```

Comment:
We should probably switch to using vectors in the parsers before we solidify this interface (so we don't need to have a breaking change later)

Reply:
This interface is experimental, so by definition any changes are not breaking.